### ISSUE TITLE
C:\ProgramData used when running in Portable mode

### ISSUE THREAD

**Issue by kees-z** [duplicati/duplicati#2826]
- [x] I have searched open and closed issues for duplicates.

----------------------------------------

## Environment info

- **Duplicati version**: 2.0.2.10 Canary
- **Operating system**: Windows 10 Enterprise
- **Backend**: N/A

## Description

When Duplicati is running in portable mode, `C:\ProgramData` is still used for downloading updates. `C:\ProgramData\Duplicati\updates` contains subfolders `2.0.2.8`, `2.0.2.9` and `2.0.2.10`.
I suppose the `updates` folder should be located in the data folder, under the Duplicati program folder, when running in portable mode.

**Comment by davegold24** [comment]
Windows standard is to not store data in program files. So technically, when running as a service, the database should be in c:\programdata\duplicati2\data, not c:\program files\duplicati2\data. 

When running in portably mode, it probably should be putting them into the %APPDATA% subtree.

**Comment by kees-z** [comment]
I agree that programs should not write to the Program files folder, but this doesn't apply for software running in portable mode.

Running in portable mode, all data should be stored at the same location as the program itself (or a subfolder). When running it from a portable device, the software should have access to all its data after connecting the portable device to another computer. Also, portable software should not leave any traces after disconnecting the storage media it runs on.

This is why software that runs in portable mode should _never_ touch system folders like %APPDATA%, %PROGRAMDATA%, the Documents and Download library etc.
Ideally, portable software should even relocate temporary files from %TEMP% to .\Temp but doing this has some major backdraws (many R/W operations to flash memory, reducing performance and storage media lifetime). So using %TEMP% is ok for me.

If you want to run Duplicati in portable mode, it should not use any system folder, including %PROGRAMDATA%. If you want to prevent storing data in the %PROGRAMFILES% (which is best practice), install Duplicati in a location outside the Program files folder, or (better option for portable installs) download and copy the Zip version to a location of your choice.

If you want to store the local databases in another location when running in portable mode, you can use `--server-datafolder` in addition to the `--portable-mode` option.

**Comment by davegold24** [comment]
@kees-z Very good points. I don't think I was aware of a --server-datafolder option.

**Comment by kenkendk** [comment]
This part is not well documented, but the logic is here:
https://github.com/duplicati/duplicati/blob/master/Duplicati/Library/AutoUpdater/UpdaterManager.cs#L92

It basically tries to find a write-able folder, and stores the updates there. When running as a service, this is a bit of a problem because "all" folders are write-able.

It does appear to attempt to avoid writing to "ProgramFiles", but maybe something else is triggering this.

**Comment by kenkendk** [comment]
Closing because the automatic update is no longer in the code.
Also, the latest canary is handling portable-mode better and will not store config files outside its installation folder.
